presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -312,7 +312,7 @@ private static Stream toStream(DwrfProto.Stream stream)
         return new Stream(
                 stream.getColumn(),
                 toStreamKind(stream.getKind()),
-                toIntExact(stream.getLength()),
```

Comment:
Instead of using toIntExactVerbose, can we do something like.

if (stream.getLength() > Integer.MAX_SIZE) {
   throw new OrcCorruptionException ("Stream Name, Stream Id is larger than supported size");
}

Reply:
Unfortunately OrcCorruptionException requires OrcDataSourceId which is not available in this method. To add it we would need to update the upstream API MetadataReader#readStripeFooter->toStream->toStream. The scope of the code change is small (<20 lines). 

Opinion? Go for it and change the API?

Reply:
We can throw any other exception, let us keep the change small. (I am thinking of IllegalStateException here or worst case, we can just throw ArithmeticException)

Reply:
Switched to OrcCorruptionException, updated API to accept the OrcDataSourceId.